### ISSUE TITLE
Possible competing caffeinate arguments 

### ISSUE THREAD

**Issue by grahampugh** [grahampugh/erase-install#170]
**Describe the bug**
The following is used to caffeinate the script:

```pid=$$
echo "   [$script_name] Caffeinating this script (pid=$pid)"
/usr/bin/caffeinate -dimsu -w $pid &
```

However, according to the manual, `-w` is ignored if `-u` is used, and `-u` effectively means "caffeinate for 5 seconds" if `-t` is not also set:

```
     -u      Create an assertion to declare that user is active. If the display is off, this option turns the
             display on and prevents the display from going into idle sleep. If a timeout is not specified with
             '-t' option, then this assertion is taken with a default of 5 second timeout.

     -t      Specifies the timeout value in seconds for which this assertion has to be valid. The assertion is
             dropped after the specified timeout. Timeout value is not used when an utility is invoked with
             this command.

     -w      Waits for the process with the specified pid to exit. Once the the process exits, the assertion is
             also released.  This option is ignored when used with utility option.
```

Recommendation: remove `-u`.

**Additional context**
This is a theoretical bug. There is one anecdotal report that it isn't working as expected, but I have not reproduced that behaviour.

**Comment by TheTechnicalSpecialist** [comment]
I'm getting this error with one of my users:

An error occurred preparing the update.
   [erase-install] 'Self Service' ended
   [erase-install] 'caffeinate' ended
/Library/Management/erase-install/erase-install.sh: line 945: 10653 Terminated: 15          /usr/bin/caffeinate -dimsu -w $pid
/Library/Management/erase-install/erase-install.sh: line 724: 10861 Killed: 9               dep_notify_progress startosinstall > /dev/null 2>&1

Is that the same issue? Any suggestions?

**Comment by grahampugh** [comment]
No, you haven't posted any error there; just the standard "kill" statements when the script finishes. Please post a new issue and follow the instructions on how to provide enough information.

**Comment by TheTechnicalSpecialist** [comment]
@grahampugh Apologies, I'm a bit new with GitHub. You've given me a clue tho. the statement 'An error occurred preparing the update.' is the error here. I'm thinking that problem is not really to do with your package/script. Thanks.

**Comment by grahampugh** [comment]
(I was incorrect in my thoughts about caffeinate - the options are already correct as they are)
